synapse/storage/push_rule.py

```diff
@@ -185,6 +185,61 @@ def bulk_get_push_rules(self, user_ids):
 
         defer.returnValue(results)
 
+    @defer.inlineCallbacks
+    def copy_push_rules_from_room_to_room(self, old_room_id, new_room_id, user_id=None):
+        """Copy the push rules from one room to another.
+
+        Args:
+            old_room_id (str): ID of the old room.
+            new_room_id (str): ID of the new room.
+            user_id (str|None): ID of user to copy push rules for. If None
+                push rules for all local users will be copied.
+        """
+        if user_id is None:
```

Comment:
when do you expect `user_id` to be None? There's a lot of code here which seems like it will never be run.